src/actions/rariActions.js

```diff
@@ -62,23 +54,28 @@ export const fetchRariUserDataAction = () => {
     if (!smartWalletAccount) return;
     const smartWalletAddress = getAccountAddress(smartWalletAccount);
 
-    dispatch({ type: SET_FETCHING_RARI_USER_DATA });
+    dispatch({ type: SET_FETCHING_RARI_DATA });
 
-    const [userDepositInUSD, userDepositInRSPT, userInterests] = await Promise.all([
+    const [userDepositInUSD, userInterests, rariAPY] = await Promise.all([
```

Comment:
Wdyt if we validate the promise output first and then destruct? Since if any of these calls fail then output is first rejection which may not be array and therefore it would crash here.